cartographer_rviz/cartographer_rviz/submaps_display.cc

```diff
@@ -256,10 +272,24 @@ void Trajectory::AllEnabledToggled() {
   }
 }
 
-Trajectory::Trajectory(std::unique_ptr<::rviz::BoolProperty> property)
+void Trajectory::PoseMarkersEnabledToggled() {
+  const bool visible = pose_markers_visibility->getBool();
+  for (auto& submap_entry : submaps) {
+    submap_entry.second->set_pose_markers_visibility(visible);
+  }
+}
+
+Trajectory::Trajectory(std::unique_ptr<::rviz::BoolProperty> property,
+                       const bool pose_markers_enabled)
     : visibility(std::move(property)) {
   ::QObject::connect(visibility.get(), SIGNAL(changed()), this,
                      SLOT(AllEnabledToggled()));
+  pose_markers_visibility = absl::make_unique<::rviz::BoolProperty>(
+      QString("Submap Pose Markers"), pose_markers_enabled,
+      QString("Toggles the submap pose markers of this trajectory."),
+      visibility.get());
```

Comment:
why visibility.get()? shouldn't this be pose_markers_enabled?

Reply:
oh yes.

Reply:
edit: no ;) it's a bit confusing, but this means that the toggle for this trajectory is in the submaps visibility list (the drop down thing I also open in the GIF)

Reply:
Added a comment to make it clear.